Find a functionally equivalent code:android.provider.Browser.truncateHistory:COMMENT
Method Modifier: public      static      final       
Comment:/**
 * If there are more than MAX_HISTORY_COUNT non-bookmark history
 * items in the bookmark/history table, delete TRUNCATE_N_OLDEST
 * of them.  This is used to keep our history table to a
 * reasonable size.  Note: it does not prune bookmarks.  If the
 * user wants 1000 bookmarks, the user gets 1000 bookmarks.
 * Requires {@link android.Manifest.permission#READ_HISTORY_BOOKMARKS}
 * Requires {@link android.Manifest.permission#WRITE_HISTORY_BOOKMARKS}
 *
 * @param cr The ContentResolver used to access the database.
 */

@@ -5,15 +5,13 @@
         // Select non-bookmark history, ordered by date
         cursor = cr.query(History.CONTENT_URI, new String[] { History._ID, History.URL, History.DATE_LAST_VISITED }, null, null, History.DATE_LAST_VISITED + " ASC");
         if (cursor.moveToFirst() && cursor.getCount() >= MAX_HISTORY_COUNT) {
-            final WebIconDatabase iconDb = WebIconDatabase.getInstance();
             /* eliminate oldest history items */
             for (int i = 0; i < TRUNCATE_N_OLDEST; i++) {
                 cr.delete(ContentUris.withAppendedId(History.CONTENT_URI, cursor.getLong(0)), null, null);
-                iconDb.releaseIconForPageUrl(cursor.getString(1));
                 if (!cursor.moveToNext())
                     break;
             }

